src/SBSecondKick.cpp

```diff
@@ -0,0 +1,507 @@
+                               double kcrit, double flux, double scale,
+                               const GSParamsPtr& gsparams) :
+        SBProfile(new SBSecondKickImpl(lam, r0, diam, obscuration, L0, kcrit, flux, scale,
+                                       gsparams)) {}
+
+    SBSecondKick::SBSecondKick(const SBSecondKick &rhs) : SBProfile(rhs) {}
+
+    SBSecondKick::~SBSecondKick() {}
+
+    double SBSecondKick::getLam() const
+    {
+        assert(dynamic_cast<const SBSecondKickImpl*>(_pimpl.get()));
+        return static_cast<const SBSecondKickImpl&>(*_pimpl).getLam();
+    }
+
+    double SBSecondKick::getR0() const
+    {
+        assert(dynamic_cast<const SBSecondKickImpl*>(_pimpl.get()));
+        return static_cast<const SBSecondKickImpl&>(*_pimpl).getR0();
+    }
+
+    double SBSecondKick::getDiam() const
+    {
+        assert(dynamic_cast<const SBSecondKickImpl*>(_pimpl.get()));
+        return static_cast<const SBSecondKickImpl&>(*_pimpl).getDiam();
+    }
+
+    double SBSecondKick::getObscuration() const
+    {
+        assert(dynamic_cast<const SBSecondKickImpl*>(_pimpl.get()));
+        return static_cast<const SBSecondKickImpl&>(*_pimpl).getObscuration();
+    }
+
+    double SBSecondKick::getL0() const
+    {
+        assert(dynamic_cast<const SBSecondKickImpl*>(_pimpl.get()));
+        return static_cast<const SBSecondKickImpl&>(*_pimpl).getL0();
+    }
+
+    double SBSecondKick::getKCrit() const
+    {
+        assert(dynamic_cast<const SBSecondKickImpl*>(_pimpl.get()));
+        return static_cast<const SBSecondKickImpl&>(*_pimpl).getKCrit();
+    }
+
+    double SBSecondKick::getScale() const
+    {
+        assert(dynamic_cast<const SBSecondKickImpl*>(_pimpl.get()));
+        return static_cast<const SBSecondKickImpl&>(*_pimpl).getScale();
+    }
+
+    double SBSecondKick::getHalfLightRadius() const
+    {
+        assert(dynamic_cast<const SBSecondKickImpl*>(_pimpl.get()));
+        return static_cast<const SBSecondKickImpl&>(*_pimpl).getHalfLightRadius();
+    }
+
+    double SBSecondKick::structureFunction(double rho) const
+    {
+        assert(dynamic_cast<const SBSecondKickImpl*>(_pimpl.get()));
+        return static_cast<const SBSecondKickImpl&>(*_pimpl).structureFunction(rho);
+    }
+
+    double SBSecondKick::kValueSlow(double k) const
+    {
+        assert(dynamic_cast<const SBSecondKickImpl*>(_pimpl.get()));
+        return static_cast<const SBSecondKickImpl&>(*_pimpl).kValueSlow(k);
+    }
+
+    double SBSecondKick::xValueSlow(double k) const
+    {
+        assert(dynamic_cast<const SBSecondKickImpl*>(_pimpl.get()));
+        return static_cast<const SBSecondKickImpl&>(*_pimpl).xValueSlow(k);
+    }
+
+    //
+    //
+    //
+    //SKInfo
+    //
+    //
+    //
+
+    SKInfo::SKInfo(double lam, double r0, double diam, double obscuration, double L0,
+                   double kcrit, const GSParamsPtr& gsparams) :
```

Comment:
>Should make lam, diam, L0, kcrit dimensionless by dividing or multiplying by r0 as appropriate.

I looked into this some, but I don't think it's possible given the combination of VonKarman and Airy-like functionality in SecondKick.  For example, the VonKarman kValue can be written as a function of k*r0, but the same isn't true for the Airy kValue.  I may have missed something though so feel free to take another look.

I didn't look into eliminating the LRUCache, though I do wonder if it may at least be helping with some of the unit tests; e.g. do_pickle.